app/client/templates/elements/compileContract.html

```diff
@@ -53,11 +53,11 @@
                             <select class="compiled-contracts">
                                 <option disabled selected>{{i18n "wallet.contracts.pickContract"}}</option>
                                 {{#each this}}
-                                    <option value="{{name}}">{{toSentence name true}}</option>
+                                    <option value="{{name}}" selected="{{selected}}">{{toSentence name true}}</option>
```

Comment:
@alexvandesande The sole existence of `selected` attribute will mark the element as selected. So instead of outputting only true/false inside the selected attribute, the return value of your helper must be the attr itself.

 Here I advise you to:
`<option value="{{name}}" {{selected}}>{{...}}</option>`

and then on helper:
`contract.selected = (contract.name == e.currentTarget.value)? 'selected' : '';`

Otherwise all elements will have the selected attribute, therefore, selecting the last element by default – the last element parsed. See this codepen for practical examples: http://codepen.io/evertonfraga/pen/YWGBON